streams/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala

```diff
@@ -529,4 +534,22 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
         }
     ) must_== Failure(Cause.Fail(e))
   }
+
+  private def bufferStream = prop { list: List[Int] =>
```

Comment:
This is an important test for verifying that buffering does not affect the stream semantics, so it's good that we have it. 

Let's now add one that exercises the point of buffering: allowing a fast producer to make progress independently of a slow consumer. This can be done by using a stream that adds elements to a `Ref[List[Int]]` before a buffer, and has a `mapM(_ => IO.never)` after the buffer. Let me know if you need any help with that.

Reply:
Thank you @iravid for the review comments. Will let you know, if any help needed.

Reply:
Fixed